### ISSUE TITLE
Add necessary permissions to example

### ISSUE THREAD

**Issue by brcourt** [aws-cloudformation/cloudformation-cli#391]
CloudWatch/Logs permissions should be added to the example. This seems to be necessary to allow the resource provider to actually write to the default Log Group for the resource. Also required for the Python plugin which uses the logger to publish logs automatically to this log-group.

At the very least, this could be specified in the documentation.
https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-develop.html#resource-type-develop-log

**Comment by tobywf** [comment]
Sorry, which example? It shouldn't be necessary, especially not on the handler permissions, since those only apply to what your handler code will do. If you use the provided logger, it should all be set up.

We create two stacks. There's a stack per resource, which includes the `ExecutionRole`. But AFAIK, the logging permission comes from the `CloudFormationManagedUploadInfrastructure` stack, which includes the `LogAndMetricsDeliveryRole` - that should have a trust relationship to allow CloudFormation to send logs to you. Doesn't that work?

**Comment by parsnips** [comment]
@tobywf I'm using the golang handler, and I do see that `LogAndMetricsDeliveryRole` created and my registration uses it:

```
{
    "Arn": "arn:aws:cloudformation:us-west-2:981211660345:type/resource/TX-Database-Schema/00000017",
    "Type": "RESOURCE",
    "TypeName": "TX::Database::Schema",
    "DefaultVersionId": "00000017",
    "IsDefaultVersion": true,
    "Description": "Schema resource for TxLayer database.",
    "Schema": "{\n  \"typeName\": \"TX::Database::Schema\",\n  \"description\": \"Schema resource for TxLayer database.\",\n  \"sourceUrl\": \"https://github.com/txlayer/aws\",\n  \"definitions\": {\n    \"Table\": {\n      \"type\": \"object\",\n      \"description\": \"A table in TxLayer\",\n      \"properties\": {\n        \"name\": {\n          \"type\": \"string\"\n        },\n        \"version\": {\n          \"type\": \"string\"\n        },\n        \"schema\": {\n          \"type\": \"object\"\n        },\n        \"policy\": {\n          \"type\": \"object\"\n        }\n      },\n      \"additionalProperties\": false\n    }\n  },\n  \"properties\": {\n    \"Table\": {\n      \"$ref\": \"#/definitions/Table\"\n    },\n    \"TableId\": {\n      \"type\": \"string\"\n    },\n    \"Tenant\": {\n      \"description\": \"The tenant identifier to run as.\",\n      \"type\": \"string\",\n      \"format\": \"uuid\"\n    }\n  },\n  \"additionalProperties\": false,\n  \"required\": [\n    \"Table\",\n    \"Tenant\"\n  ],\n  \"readOnlyProperties\": [\n    \"/properties/TableId\"\n  ],\n  \"primaryIdentifier\": [\n    \"/properties/TableId\"\n  ],\n  \"handlers\": {\n    \"create\": {\n      \"permissions\": [\n      ]\n    },\n    \"read\": {\n      \"permissions\": [\n      ]\n    },\n    \"update\": {\n      \"permissions\": [\n      ]\n    },\n    \"delete\": {\n      \"permissions\": [\n      ]\n    },\n    \"list\": {\n      \"permissions\": [\n      ]\n    }\n  }\n}\n",
    "ProvisioningType": "FULLY_MUTABLE",
    "DeprecatedStatus": "LIVE",
    "LoggingConfig": {
        "LogRoleArn": "arn:aws:iam::981211660345:role/CloudFormationManagedUplo-LogAndMetricsDeliveryRol-UMOIGZ0FE0RY",
        "LogGroupName": "tx-database-schema-logs"
    },
    "ExecutionRoleArn": "arn:aws:iam::981211660345:role/tx-database-schema-role-stack-ExecutionRole-3O7JEZQA6PDP",
    "Visibility": "PRIVATE",
    "SourceUrl": "https://github.com/txlayer/aws",
    "LastUpdated": "2020-12-10T22:00:19.176000+00:00",
    "TimeCreated": "2020-12-10T00:20:07.097000+00:00"
}
```

However whenever I create a stack that uses that resource,  I do not see any logs other than the "Stream Initialized" log. Is there _something else_  we need to do to get logging flowing?

**Comment by tobywf** [comment]
Sorry, I no longer work for AWS, so I'm not up-to-date on this. I'm sure someone from the team will get back to you shortly! In the meantime, I'm not sure if you care about this or not, but you may want to scrub any identifying info from the snippet.

**Comment by parsnips** [comment]
@tobywf thanks for the response :) We tracked it down to the logging provider being controlled through build tags and defaulting to a no-op implementation ... I think we figured out!

**Comment by jamestelfer** [comment]
Also using the golang handler, and I got stuck on the same thing: no logging unless I specified the `logging` tag to the build.

The generated `makebuild` makefile has the following:
```Makefile
	env GOOS=linux go build -ldflags="-s -w" -tags="$(TAGS)" -o bin/handler cmd/main.go
```

Invoking the build setting the `TAGS` var makes it work:
```sh
TAGS=logging make
```

Thanks @parsnips for the pointer in the right direction!

This wasn't obvious from the doco, and took a while to figure out -- I'll raise an issue on the Go plugin to see if there is a good place to document this.
